Tasks/Common/kubernetes-common-v2/helmutility.ts

```diff
@@ -66,16 +67,32 @@ function getHelmDownloadURL(version: string): string {
 
 export async function getStableHelmVersion(): Promise<string> {
     try {
-        const downloadPath = await toolLib.downloadTool(helmLatestReleaseUrl);
-        const response = JSON.parse(fs.readFileSync(downloadPath, 'utf8').toString().trim());
-        if (!response.tag_name)
-        {
-            return stableHelmVersion;
-        }
-        
-        return response.tag_name;
+        const downloadPath = await toolLib.downloadTool(helmAllReleasesUrl);
+        const responseArray = JSON.parse(fs.readFileSync(downloadPath, 'utf8').toString().trim());
+        let latestHelmVersion = semver.clean(stableHelmVersion);
+        responseArray.forEach(response => {
+            if (response && response.tag_name) {
+                let currentHelmVerison = semver.clean(response.tag_name.toString());
+                if (currentHelmVerison) {
+                    if (currentHelmVerison.toString().indexOf('rc') == -1 && semver.gt(currentHelmVerison, latestHelmVersion)) {
+                        //If current helm version is not a pre release and is greater than latest helm version
+                        latestHelmVersion = currentHelmVerison;
+                    }
+                }
+            }
+        });
+        latestHelmVersion = "v" + latestHelmVersion;
+        return latestHelmVersion;
     } catch (error) {
-        tl.warning(tl.loc('HelmLatestNotKnown', helmLatestReleaseUrl, error, stableHelmVersion));
+        let telemetry = {
+            error: tl.loc("HelmLatestNotKnown", helmAllReleasesUrl, error, stableHelmVersion)
```

Comment:
It doesn't have to be a loc string.

Reply:
This can be a generic JSON. So, you can split this into multiple parts.
Something like.
```json
{
   "event": "GitHubReleaseFetchFailed",
   "error": error
}
```

Reply:
Updated.